src/data/equiv/basic.lean

```diff
@@ -1906,6 +1906,15 @@ begin
   rw swap_apply_of_ne_of_ne hi hj,
 end
 
+lemma swap_apply_eq_swap_apply_iff [decidable_eq α] {x y z w : α} :
+  swap x y z = w ↔ swap x y w = z :=
```

Comment:
This goal is a bit more like many of the equiv lemmas for this type of thing, and can probably be solved with two or three rewrites:
```suggestion
  swap x y z = w ↔  z = swap x y w :=
```
Better or worse in your opinion?

Reply:
The goal I use this for looks like:
```
α: Type u_1
_inst_1: decidable_eq α
f: perm α
xya: α
ha: ⇑f a ≠ a
H: ⇑(swap x y) (⇑f a) = a
hx: a ≠ x
hy: a ≠ y
⊢ false
```
with a tactic proof of 
```
  rw [swap_apply_eq_swap_apply_iff, swap_apply_of_ne_of_ne hx hy] at H,
  exact ha H.symm
```
Your suggestion would remove the necessity for `H.symm` and make it `H`.